### PR TITLE
refactor: update goal_code invalid name for OOB V2

### PR DESCRIPTION
**Pull request by Baha-sk**
Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: pkg/wallet/invitation.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type GenericInvitation struct {
 	Type      string                        `json:"type"`
 	Label     string                        `json:"label,omitempty"`
 	Goal      string                        `json:"goal,omitempty"`
-	GoalCode  string                        `json:"goal_code,omitempty"`
+	GoalCode  string                        `json:"goal-code,omitempty"`
```
Comment (sudeshrshetty) [inline pkg/wallet/invitation.go:41]: This one will break current DIDComm V1 support for wallet

**Review by sudeshrshetty — CHANGES_REQUESTED** [review]

File: pkg/wallet/invitation.go (hunk 1 above)
Comment (Baha-sk) [reply pkg/wallet/invitation.go:41]: Then we need 2 distinct models of `GenericInvitation` and `rawInvitation` and separate V1 from V2

Comment (Baha-sk) [reply pkg/wallet/invitation.go:41]: actually, this doesn't break V1 support, notice ID and Type above are V2 as well. The logic of flipping between V1 and V2 is used elsewhere. No additional change is needed here.

Comment (sudeshrshetty) [reply pkg/wallet/invitation.go:41]: but it is used as request model in command controller here. It comes down to problem we discussed yesterday where command controller API doc shows different model than the one command is expecting.

https://github.com/hyperledger/aries-framework-go/blob/main/pkg/controller/command/vcwallet/models.go#L351

Created [issue](https://github.com/hyperledger/aries-framework-go/issues/3106) for that, and this one can be merged, but need to keep an eye.

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**